Update scripts and container to up-to-date CDS API

trossi opened the pull request:
This PR will update scripts and container to up-to-date CDS API.

Old API keys are not valid anymore and no `API_User` is needed for the new API. Container image version 0.4 includes [ecmfwr version 2](https://github.com/bluegreen-labs/ecmwfr) and patched KrigR installed from [this branch](https://github.com/trossi/KrigR/tree/update-cdsapi).

evalieungh commented:
@ErikKusch and @trossi is this ready to merge? I can resolve the conflicts and complete the merge if that's OK with you. But I'm not confident enough that I understand or can test all the changes.

trossi commented:
@evalieungh I'll push updates to containers so that they match with Erik's changes.

evalieungh commented:
Great, @trossi. I took the chance of resolving the conflicts, hopefully correctly

trossi commented:
@evalieungh Thanks for doing the merge. I updated container scripts and added some minor fixes.

I noticed there is now `API_User` back in `SHARED-Data.R`, but API_User doesn't exists in the new API. @ErikKusch Could these be removed or does KrigR need it?

ErikKusch commented:
Still required I am afraid

trossi commented:
OK, I added then back also the checks that some API_User is defined 15b549d4aea8e3c08dab8771d41d2f6d1ae3af35.

@evalieungh I have done my changes, but I haven't run any tests after these latest changes. Would you like to test?

evalieungh commented:
I'm not sure how to test it. I'm not planning to use the ModGP but trying to make a parallel incorporation of CAPFITOGEN at the moment

trossi commented:
I run `ModGP-run_prep.R` for testing and pushed small fixes.

The call `Qsoil1_ras <- CDownloadS(..)` fails in KrigR with:
```
...
###### Data Checking & Limitting & Aggregating
[1] "Checking for known data issues"
[1] "Spatial Limiting"
[1] "Temporal Aggregation"
###### Data Export & Return
Error in ncdf4::ncvar_put(ncobj, ncvars[[i]], d, start = c(1, b$row[j],  : 
  long vectors not supported yet: ../../src/include/Rinlinedfuns.h:537
Calls: FUN.DownBV ... writeCDF -> writeCDF -> .local -> .write_cdf -> <Anonymous>
In addition: Warning messages:
1: In Meta.Check(DataSet = DataSet, Type = Type, VariableCheck = QueryVariable,  :
  Cannot validate user-specified end date (DateStop) because specified data set is being updated regularly (Lag = 3 month). User-specification may lead to an error.
2: [mask] CRS do not match 
Execution halted
```

@ErikKusch Might you know a fix for this?

ErikKusch commented:
That is curious. I cannot recreate this issue on my end. What version of KrigR is in the docker?

trossi commented:
The container has KrigR 0.9.4, but the odd thing is that I can't reproduce this issue on my local computer either (although using the same container). So, the issues seems to be more delicate and needs more debugging. I think this PR could be merged though so it's not blocking @evalieungh  work.

evalieungh commented:
OK, then we take the chance of merging the PR despite some issues. If there are issues down the line we can deal with it then.

evalieungh commented:
@trossi OK to delete the update-cdsapi branch as well or should we keep it for debugging the KrigR issue?

trossi commented:
> OK to delete the update-cdsapi branch as well or should we keep it for debugging the KrigR issue?

Branch can be deleted. I'll open a new branch and PR for the fix (I found the issue but do a bit testing still).
